.github/actions/check-compatible-swift/action.yml

```
@@ -29,11 +29,24 @@ runs:
             local minor="$(echo $1 | cut -d . -f 2)"; \
             local patch="$(echo $1 | cut -d . -f 3)"; \
             \
-            echo $(( ${major} * 10000 + ${minor} * 100 + ${patch} )); \
+            echo $(( ${major} * 10000 + ${minor:-0} * 100 + ${patch:-0} )); \
           }
           
           toolchain_version=$(parse_version $(swift package --version | cut -c31-))
           package_version=$(parse_version $(swift package ${PACKAGE_ROOT} tools-version))
           
-          is_compatible=$((( ${package_version} <= ${toolchain_version} )) && echo true || echo false)
-          echo "ok=${is_compatible}" >>"${GITHUB_OUTPUT}"
+          swift_is_compatible=$((( ${package_version} <= ${toolchain_version} )) && echo true || echo false)
+
+          if [ "$(uname)" = "Darwin" ]; then
+            platform_version=$(parse_version $(swift package dump-package | jq -r '.platforms[] | select(.platformName == "macos") | .version'))
+            if [ "${platform_version}" = "null" ] || [ -z "${platform_version}" ]; then
+              platform_is_compatible=true
+            else
+              current_platform=$(parse_version $(sw_vers -productVersion))
+              platform_is_compatible=$((( ${platform_version} <= ${current_platform} )) && echo true || echo false)
+            fi
+          else
+            platform_is_compatible=true
+          fi
+
+          echo "ok=$([ "${swift_is_compatible}" = true ] && [ "${platform_is_compatible}" = true ] && echo true || echo false)" >> "${GITHUB_OUTPUT}"
```

Could you at least use modern Bash syntax here?
```suggestion
          if [[ "$(uname)" == "Darwin" ]]; then
            platform_version=$(parse_version $(swift package dump-package | jq -r '.platforms[] | select(.platformName == "macos") | .version'))
            if [[ "${platform_version}" == "null" || -z "${platform_version}" ]]; then
              platform_is_compatible=true
            else
              current_platform=$(parse_version $(sw_vers -productVersion))
              platform_is_compatible=$((( ${platform_version} <= ${current_platform} )) && echo true || echo false)
            fi
          else
            platform_is_compatible=true
          fi

          echo "ok=$([[ "${swift_is_compatible}" == true && "${platform_is_compatible}" == true ]] && echo true || echo false)" >> "${GITHUB_OUTPUT}"
```

---

Sorry for trying to keep up with your old code 😛